app/models/user.rb

```diff
@@ -49,6 +49,9 @@ class User < ActiveRecord::Base
   has_many :votes, dependent: :destroy
   has_many :voted_events, through: :votes, source: :events
   has_many :subscriptions, dependent: :destroy
+
+  has_and_belongs_to_many :public_events, class_name: 'Event'
```

Comment:
Why public_events? The name doesn't really make sense for me?
Aren't they ``favorite_events``?